### ISSUE TITLE
KuCoin spot stop loss doesn't work

### ISSUE THREAD

**Issue by hodlerhacks** [ccxt/ccxt#18606]
### Operating System

Windows

### Programming Languages

JS

### CCXT Version

4.0.32

### Description

Stop loss orders, using the unified method, on KuCoin doesn't work. As an example, this is what I did:

await exchange.createOrder('SOL/USDT', 'limit', 'sell', 0.0869, 26.5, { stopLossPrice: 26.6 });

The SOL/USDT price was around 26.85 when I created the order. The order was immediately triggered and filled. Instead, it should only be triggered when the price drops below 26.6.

The same code works well with Binance.

Now, I tried the above 3 times in a row. And interestingly, the 2nd time it did work as expected. I then manually canceled the stop loss, and ran the exact same code again. And the SOL was immediately sold.

Any idea what could be the problem?

### Code

```
  

```

**Comment by carlosmiei** [comment]
Hello @hodlerhacks, can you confirm if Kucoin displays the order type+order+price+sl price correctly in the WEB UI? If so I would say the order is being correctly "created and sent" in CCXT.

**Comment by hodlerhacks** [comment]
I'm currently traveling and haven't been able to access the order history, as I placed the orders on a sub account. (I'm new to KuCoin, and was only able to find the order history for the main account.) I could find the orders in Already, and can confirm those were stop loss limit orders. When I'm back 'online', I'll try to figure out the price levels used.

**Comment by hodlerhacks** [comment]
Ok, I managed to find the order details on KuCoin. Here's one example:

<html>
<body>


Time | Pair | Side | Type | Trigger Condition | Order Price | Amount | Filled | Avg. Price | Status
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
  |   |   |   |   |   |   |   |   |  
2023/07/20 09:22:20 | SOL/USDT | Sell | Stop Limit Order | ≥26.6USDT | 26.5USDT | 0.0869SOL | 0.0869SOL | 26.865USDT | Completed


</body>
</html>

It looks like the trigger condition is not set correctly. The order should have been triggered when the price drops below 26.6.

Is this a bug in CCXT or on the KuCoin side?

**Comment by hodlerhacks** [comment]
By the way, I guess the reason why it seemed to work for one of the orders was that that order didn't immediately get filled, so I saw the stop loss limit appear on the chart (in Altrady), and then I canceled it.

And, I guess this is where the problem is (line 1833):

            request['stop'] = isStopLoss ? 'entry' : 'loss';

**Comment by hodlerhacks** [comment]
@ttodua, I noticed you added the 'enhancement' label. Isn't this a bug? I don't mind the label really, but I can imagine it may be picked up with a different priority because of it. ;-)

**Comment by ttodua** [comment]
@hodlerhacks you are welcome to mention your thoughts !
actually, it can be also labeled as a bug, but per se, to say,
A) if something is not implemented, and we have to add that functionality, we do an "_ **enhancement**_"
B) if something is implemented, but does not work as expected and there is some mistake we have to fix a "_**bug**_".
but sure, I don't mind labeling this a a bug, as it somehow related..

**Comment by hodlerhacks** [comment]
@ttodua, got it, thanks for clarifying, and for adding the 'bug' label :-)

**Comment by ttodua** [comment]
I have checked and kucoin API does not support embedding SL & TP orders, so even if you use ccxt's unified params, it is not used in api at all. we can't do anything here, you might push that to kucoin API support and ask them that they implemented the feature so it was possible through API

**Comment by hodlerhacks** [comment]
@ttodua, thanks for looking into this. The thing is, when I send the order as follows:

await exchange.createOrder('SOL/USDT', 'limit', 'sell', 0.0869, 26.5, { stopLossPrice: 26.6 });

It does place a stop limit order on KuCoin, I confirmed that in the order details on KuCoin. The only thing that doesn't go well is setting the trigger condition. It is set to ≥26.6USDT, while it should be <26.6USDT. And that can be explained by the following code in CCXT:

`request['stop'] = isStopLoss ? 'entry' : 'loss';`

This is not correct, right? I believe it should be:

`request['stop'] = isStopLoss ? 'loss' : 'entry';`

And then I would expect the trigger condition to be set properly. Does that make sense?

**Comment by ttodua** [comment]
@hodlerhacks
sorry, I think I was confused. So, you meant a separate "reducy-only" order (stop-loss, take-profit.. etc are just generic names) so we call all of them trigger orders, so whenever price reaches activation price, the order gets triggered.
my bad, I now understood that you meant trigger (stoploss) order. I'll re-open the topic and answer your issue a bit later

**Comment by hodlerhacks** [comment]
Hi @ttodua, I'm not sure if I mean a reduce-only order (I'm not very familiar with term). Isn't that mostly relevant to future trading? In this case, I'm placing orders on the spot market. But indeed, what I'm trying to achieve is a sell order that is triggered when the price drops below a certain price, using the unified CCXT implementation as described in this section: https://github.com/ccxt/ccxt/wiki/Manual#stop-loss-orders. So it seems we're talking about the same. Thanks for re-opening the issue. :-)

**Comment by ttodua** [comment]
@hodlerhacks I identified the issue you had reported, and submited a fix PR. once that PR is merged, this issue will be close, update your CCXT and it will work correctly.
thanks for reporting the issue

**Comment by hodlerhacks** [comment]
Excellent, thanks for the quick fix @ttodua!

**Comment by ttodua** [comment]
> I've tested and indeed stop limit orders now work, so that's great. However, I've noticed that when placing or canceling these stop limit orders on KuCoin, those events are not caught by the watchOrders() method. I've validated the exact same code with Binance, and then it works without problems. The problem is 100% reproducible on KuCoin.
> 
> I figured, I'll report it in this thread. But if you want me to create a new issue, let me know.

better to open a separate/new issue for different/WS problem 👍

**Comment by hodlerhacks** [comment]
Done: https://github.com/ccxt/ccxt/issues/18726
